### PR TITLE
GraphQL: Project Type

### PR DESCRIPTION
**Pull request by ericenns**
## What does this PR do and why?
_Describe in detail what your merge request does and why._

This PR adds a Project type to the GraphQL api which implements the Relay Node interface. 

## Screenshots or screen recordings
_Screenshots are required for UI changes, and strongly recommended for all other pull requests._

## How to set up and validate locally
_Numbered steps to set up and validate the change are strongly suggested._

1. Login as `admin@email.com`
1. Open the graphiql interface `http://localhost:3000/graphiql`
1. Query for a Project that you can access.
   * ```graphql
     {
       project(fullPath: "bacillus/bacillus-anthracis/outbreak-2022") {
         id
         name
         path
       }
     }
     ```
1. See that you get a valid result
1. Query for a Project that you cannot access.
   * ```graphql
     {
       project(fullPath: "borrelia/borrelia-burgdorferi/outbreak-2022") {
         id
         name
         path
       }
     }
     ```
1. See that you get an authorization error

## PR acceptance checklist
This checklist encourages us to confirm any changes have been analyzed to reduce risks in quality, performance, reliability, security, and maintainability.

- [x] I have evaluated the [PR acceptance checklist](https://phac-nml.github.io/irida-next/docs/development/development_processes/code_review#acceptance-checklist) for this PR.

### CODE REVIEW

**Review by JeffreyThiessen — DISMISSED** [review]
Code looks good and everything looks to be working as intended.

**Review by deepsidhu85 — CHANGES_REQUESTED** [review]
This looks good. Just a few comments below for the `description` of a project and namespace

File: app/graphql/types/namespace.rb [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+
+module Types
+  # Namespace interface
+  module Namespace
+    include Types::BaseInterface
+    # Add the `id` field
+    include GraphQL::Types::Relay::NodeBehaviors
+    description 'A namespace'
+
+    field :description, String, null: false, description: 'Description of the namespace.'
```
[the remark this answers is not in the document]
Comment (ericenns) [reply app/graphql/types/namespace.rb]: fixed in eeb311f

File: app/graphql/types/project_type.rb [hunk 2]
```diff
@@ -1,16 +1,18 @@
...
     implements GraphQL::Types::Relay::Node
-    description 'A namespace'
+    description 'A project'
 
-    field :description, String, null: false, description: 'Description of the namespace.'
-    field :full_name, String, null: false, description: 'Full name of the namespace.'
-    field :full_path, ID, null: false, description: 'Full path of the namespace.' # rubocop:disable GraphQL/ExtractType
-    field :name, String, null: false, description: 'Name of the namespace.'
-    field :path, String, null: false, description: 'Path of the namespace.'
+    field :description, String, null: false, description: 'Description of the project.'
```
[the remark this answers is not in the document]
Comment (ericenns) [reply app/graphql/types/project_type.rb]: fixed in eeb311f

[the remark this answers is not in the document]
Comment (deepsidhu85) [reply app/graphql/types/project_type.rb]: Thanks!

File: app/graphql/types/namespace.rb (hunk 1 above)
[the remark this answers is not in the document]
Comment (deepsidhu85) [reply app/graphql/types/namespace.rb]: Thanks!

**Review by deepsidhu85 — APPROVED** [review]

**Pull request merged**